File: clientApp/src/app/models/revenue.ts
export interface Revenue {
  id: string;
  saleDate: Date;
  cultivatedWeight: number;
  soldWeight: number;
  salePrice: number;
  sampleWeight: number;
  donateWeight: number;
  wastage: number;
  remark: string;
  reconciled: boolean;
  addedByUserName: string;
  addedDate: Date;
}

export class RevenueAddValues {
  id?: string = undefined;
  saleDate: Date = new Date();
  cultivatedWeight: number | string = '';
  soldWeight: number | string = '';
  salePrice: number = 5;
  sampleWeight: number = 0;
  donateWeight: number = 0;
  wastage: number = 0;
  remark: string = '';

  constructor(revenue?: RevenueAddValues) {
    if (revenue) {
      this.id = revenue.id;
      this.saleDate = revenue.saleDate;
      this.cultivatedWeight = revenue.cultivatedWeight;
      this.soldWeight = revenue.soldWeight;
      this.salePrice = revenue.salePrice;
      this.sampleWeight = revenue.sampleWeight;
      this.donateWeight = revenue.donateWeight;
      this.wastage = revenue.wastage;
      this.remark = revenue.remark;
    }
  }
}

File: Domain/Revenue.cs
using System.ComponentModel.DataAnnotations;

namespace Domain;

public class Revenue
{
    public Guid Id { get; set; }
    public DateTime SaleDate { get; set; }
    public int CultivatedWeight { get; set; }
    public int SoldWeight { get; set; }
    public int SalePrice { get; set; }
    public int SampleWeight { get; set; }
    public int DonateWeight { get; set; }
    public int Wastage { get; set; }    //To be deleted
    [StringLength(250)]
    public string Remark { get; set; }
    /// <summary>
    /// True if Sale weight is matching with sum of sale register 
    /// </summary>
    public bool Reconciled { get; set; }
    public AppUser AddedBy { get; set; }
    public DateTime AddedDate { get; set; } = DateTime.UtcNow;
}
